### PR TITLE
chore: add build test base Github action implementation

### PR DESCRIPTION
**Pull request by davidgamez**
**Summary:**

- Set sqlacodegen to 2.3.0post1
- Set the Python version to 3.10
- Add GH actions to build and test. This action generates the DB and API-related files
- Modified `db-gen.sh` script to exit with code `1` if any warning or error is raised. This is the case of geometry fields with Python 3.11, failing build examples: https://github.com/MobilityData/mobility-feed-api/actions/runs/5861754004/job/15892384951?pr=78 and https://github.com/MobilityData/mobility-feed-api/actions/runs/5861777991/job/15892450794

**Expected behavior:** 

Build generates all the DB and API code, and no errors are raised.

Please make sure these boxes are checked before submitting your pull request - thanks!

- [ ] Run the unit tests with `./scripts/api-tests.sh` to make sure you didn't break anything
- [ ] Format the title like "feat: [new feature short description]". Title must follow the Conventional Commit Specification(https://www.conventionalcommits.org/en/v1.0.0/).
- [ ] Linked all relevant issues
- [ ] Include screenshot(s) showing how this pull request works and fixes the issue(s)

### CODE REVIEW

**Review by qcdyx — APPROVED** [review]
LGTM

File: .github/workflows/build-test.yml [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+          echo "POSTGRES_USER=${{ env.local_postgres_user }}" > config/.env.local
+          echo "PGUSER=${{ env.local_postgres_user }}" >> config/.env.local
+          echo "POSTGRES_PASSWORD=${{ env.local_postgres_pwd }}" >> config/.env.local
+          echo "POSTGRES_DB=${{ env.local_postgres_db }}" >> config/.env.local
+          echo "POSTGRES_PORT=${{ env.local_postgres_port }}" >> config/.env.local
+          echo "POSTGRES_HOST=localhost" >> config/.env.local
+          echo "ENV=dev" >> config/.env.local
+
+      - name: Docker Compose DB/Liquibase for db-gen.sh
+        run: docker-compose --env-file ./config/.env.local  -f docker-compose-liquibase.yaml up -d
```
Comment (cka-y) [inline .github/workflows/build-test.yml:45]: is there a reason why we need an extra docker-compose file and cannot use `docker-compose --env-file ./config/.env.local  -f up -d liquibase` to generate an instance of the database and run the liquibase update exclusively?

Comment (davidgamez) [reply .github/workflows/build-test.yml:45]: To instantiate the Postgres DB locally.

File: README.md [hunk 2]
```diff
@@ -16,7 +16,7 @@ Folder `api` contains source code of the API implementation. This repository rel
 
 ## Requirements.
 
-Python == 3.11
+Python <= 3.10
```
Comment (jcpitre) [inline README.md:19]: It would be good to explain in the file why we need to limit the Python version.

File: scripts/db-gen.sh [hunk 3]
```diff
@@ -16,8 +16,28 @@ ENV_PATH=$SCRIPT_PATH/../config/.env.local
 source "$ENV_PATH"
 rm -rf "$SCRIPT_PATH/../api/src/database_gen/"
 mkdir "$SCRIPT_PATH/../api/src/database_gen/"
-pip3 install -r "${SCRIPT_PATH}/../api/requirements.txt"
-sqlacodegen "postgresql://${POSTGRES_USER}:${POSTGRES_PASSWORD}@${POSTGRES_HOST}:${POSTGRES_PORT}/${POSTGRES_DB}" --outfile "${OUT_FILE}"
+pip install -r "${SCRIPT_PATH}/../api/requirements.txt"
```
Comment (jcpitre) [inline scripts/db-gen.sh:19]: Just curious, why not pip3?

File: README.md (hunk 2 above)
Comment (davidgamez) [reply README.md:19]: Documentation added!

**Review by jcpitre — APPROVED** [review]

**Pull request merged**